app/src/main/java/com/example/dogbook/main/viewholders/PostViewHolder.java

```diff
@@ -138,10 +147,21 @@ public void onFailure(Post post) {
             return;
         }
 
+        //Go to User Details Fragment
+        if (v == tvUsername || v == tvOwner || v == ivProfilePicture) {
+            fragmentManager.beginTransaction()
+                    .replace(R.id.timelineFragmentContainer, ProfileDetailsFragment.newInstance((User) post.getAuthor()))
+                    .addToBackStack(null)
+                    .commit();
+            return;
+        }
 
         //Add click listener to the entire item only if it is in the Posts Adapter
-        if (adapter instanceof PostsAdapter && ((PostsAdapter) adapter).clickListener != null) {
-            ((PostsAdapter) adapter).clickListener.onItemClick(getAdapterPosition(),v);
+        if (adapter instanceof PostsAdapter) {
+            fragmentManager.beginTransaction()
+                    .replace(R.id.timelineFragmentContainer, PostDetailsFragment.newInstance(post))
+                    .addToBackStack(null)
+                    .commit();
```

Comment:
Alternatively, you could have 2 separate click listeners -- one which takes you profile and one which takes you to the post details.  I don't really feel strongly either way, though

Reply:
^then you just set the 2 separate click listeners on the views, though it wouldn't necessarily cover it if you wanted to have it such that clicking on the post background took you to the details